core/src/main/java/com/predic8/membrane/core/interceptor/misc/SetHeaderInterceptor.java

```diff
@@ -72,7 +83,10 @@ public Outcome handleResponse(Exchange exc) throws Exception {
     }
 
     private Outcome handleMessage(Exchange exchange, Message msg) {
-        msg.getHeader().setValue(name, evaluateExpression(new ExchangeEvaluationContext(exchange, msg).getStandardEvaluationContext()));
+        if (!ifAbsent || !msg.getHeader().contains(name)) {
```

Comment:
Can you reverse the condition and make it as a guard? It is hard to understand unterwise.

Reply:
but then its like

```java
if (bool) return CONTINUE;
setStuff()...
return CONTINUE;
```

and the function always and multiple times returns CONTINUE

Reply:
changed to guard